File: m1-extra-exercises-easy/src/main/java/com/techelevator/Exercises.java
package com.techelevator;

import java.util.ArrayList;
import java.util.Collections;
import java.util.LinkedList;
import java.util.List;

public class Exercises {

	/*
	 1. Given an array of ints, return the number of 9's in the array.
	 arrayCount9([1, 2, 9]) → 1
	 arrayCount9([1, 9, 9]) → 2
	 arrayCount9([1, 9, 9, 3, 9]) → 3
	 */
	public int arrayCount9(int[] nums) {
		int count = 0;

		for (int i = 0; i < nums.length; i++){
			if (nums[i]== 9)
				count++;
		}

		return count;
	}

	/*
	 2. Given an array of ints, return true if one of the first 4 elements in the array is a 9. The array length may be less than 4.
	 arrayFront9([1, 2, 9, 3, 4]) → true
	 arrayFront9([1, 2, 3, 4, 9]) → false
	 arrayFront9([1, 2, 3, 4, 5]) → false
	 */
	public boolean arrayFront9(int[] nums) {

		int len = (nums.length < 4) ? nums.length : 4;
		for (int i = 0; i < len; i++) {
			if (nums[i] == 9) return true;
		}
		return false;
	}

	/*
	 3. Given an array of ints, return true if .. 1, 2, 3, .. appears in the array somewhere.
	 array123([1, 1, 2, 3, 1]) → true
	 array123([1, 1, 2, 4, 1]) → false
	 array123([1, 1, 2, 1, 2, 3]) → true
	 */
	public boolean array123(int[] nums) {

		for (int i =0; i < nums.length -2; i ++){
			if (nums[i] == 1 && nums[i+1] == 2 && nums[i+2] == 3) return true;
		}

		return false;
	}
	
	/*
	 4. Given an array of ints, we'll say that a triple is a value appearing 3 times in a row in the array. 
	 Return true if the array does not contain any triples.
	 noTriples([1, 1, 2, 2, 1]) → true
	 noTriples([1, 1, 2, 2, 2, 1]) → false
	 noTriples([1, 1, 1, 2, 2, 2, 1]) → false
	 */
	public boolean noTriples(int[] nums) {

		for (int i = 0; i < nums.length-2; i++){
			if (nums[i] == nums[i+1] && nums[i] == nums[i+2]) return false;
		}
		return true;
	}

	/*
	 5. Given an array of ints, return a new array length 2 containing the first and last elements from the 
	 original array. The original array will be length 1 or more.
	 makeEnds([1, 2, 3]) → [1, 3]
	 makeEnds([1, 2, 3, 4]) → [1, 4]
	 makeEnds([7, 4, 6, 2]) → [7, 2]
	 */
	public int[] makeEnds(int[] nums) {
		return new int[]{nums[0], nums[nums.length-1]};
	}

	/*
	 6. Given an int array length 2, return true if it contains a 2 or a 3.
	 has23([2, 5]) → true
	 has23([4, 3]) → true
	 has23([4, 5]) → false
	 */
	public boolean has23(int[] nums) {

		for (int i =0; i < nums.length;i++){
			if (nums[i] == 2 || nums[i] == 3) return true;
		}
		return false;
	}

	/*
	 7. Given an int array length 2, return true if it does not contain a 2 or 3.
	 no23([4, 5]) → true
	 no23([4, 2]) → false
	 no23([3, 5]) → false
	 */
	public boolean no23(int[] nums) {

		for (int i = 0; i < nums.length; i++){
			if (nums[i] == 2 || nums[i] == 3) return false;
		}
		return true;
	}

	/*
	 8. Given an int array, return a new array with double the length where its last element is the same as the 
	 original array, and all the other elements are 0. The original array will be length 1 or more. Note: by 
	 default, a new int array contains all 0's.
	 makeLast([4, 5, 6]) → [0, 0, 0, 0, 0, 6]
	 makeLast([1, 2]) → [0, 0, 0, 2]
	 makeLast([3]) → [0, 3]
	 */
	public int[] makeLast(int[] nums) {
		int[] output = new int[nums.length * 2];

		output[output.length-1] = nums[nums.length-1];

		return output;
	}

	/*
	 9. Given an int array, return true if the array contains 2 twice, or 3 twice. The array will be length 0, 1, or 2.
	 double23([2, 2]) → true
	 double23([3, 3]) → true
	 double23([2, 3]) → false
	 */
	public boolean double23(int[] nums) {
		int count3 = 0;
		int count2 = 0;

		for (int i = 0; i < nums.length; i++){
			if (nums[i] == 2) count2++;
			if (nums[i] == 3) count3++;
		}
		return (count2 == 2 || count3 == 2);
	}

	/*
	 10. Given an int array length 3, if there is a 2 in the array immediately followed by a 3, set the 3 element to 0. 
	 Return the changed array.
	 fix23([1, 2, 3]) → [1, 2, 0]
	 fix23([2, 3, 5]) → [2, 0, 5]
	 fix23([1, 2, 1]) → [1, 2, 1]
	 */
	public int[] fix23(int[] nums) {

		for (int i = 0; i < nums.length-1; i++){
			if (nums[i] == 2 && nums[i+1] ==3) nums[i+1] = 0;
		}
		return nums;
	}

	/*
	 11. Start with 2 int arrays, a and b, of any length. Return how many of the arrays have 1 as their first element.
	 start1([1, 2, 3], [1, 3]) → 2
	 start1([7, 2, 3], [1]) → 1
	 start1([1, 2], []) → 1
	 */
	public int start1(int[] a, int[] b) {
		int count = 0;

		if (a.length >= 1 && a[0] == 1) count++;
		if (b.length >= 1 && b[0] == 1) count++;

		return count;
	}

	/*
	 12. Start with 2 int arrays, a and b, each length 2. Consider the sum of the values in each array. Return the 
	 array which has the largest sum. In event of a tie, return a.
	 biggerTwo([1, 2], [3, 4]) → [3, 4]
	 biggerTwo([3, 4], [1, 2]) → [3, 4]
	 biggerTwo([1, 1], [1, 2]) → [1, 2]
	 */
	public int[] biggerTwo(int[] a, int[] b) {
		int sumA = 0;
		int sumB = 0;

		for (int i = 0; i < a.length; i++) {
			sumA +=a[i];
		}
		for (int i = 0; i < b.length; i++) {
			sumB += b[i];
		}

		if (sumA >= sumB) return a;

		return b;
	}
	
	/*
	 13. Given an array of ints of even length, return a new array length 2 containing the middle two elements from 
	 the original array. The original array will be length 2 or more.
	 makeMiddle([1, 2, 3, 4]) → [2, 3]
	 makeMiddle([7, 1, 2, 3, 4, 9]) → [2, 3]
	 makeMiddle([1, 2]) → [1, 2]
	 */
	public int[] makeMiddle(int[] nums) {
		int len = nums.length;

		return new int[]{nums[(len/2)-1],nums[len/2]};
	}

	/*
	 14. Given 2 int arrays, each length 2, return a new array length 4 containing all their elements.
	 plusTwo([1, 2], [3, 4]) → [1, 2, 3, 4]
	 plusTwo([4, 4], [2, 2]) → [4, 4, 2, 2]
	 plusTwo([9, 2], [3, 4]) → [9, 2, 3, 4]
	 */
	public int[] plusTwo(int[] a, int[] b) {
		return new int[]{a[0], a[1], b[0], b[1]};
	}

	/*
	 15. Given an array of ints, swap the first and last elements in the array. Return the modified array. The array 
	 length will be at least 1.
	 swapEnds([1, 2, 3, 4]) → [4, 2, 3, 1]
	 swapEnds([1, 2, 3]) → [3, 2, 1]
	 swapEnds([8, 6, 7, 9, 5]) → [5, 6, 7, 9, 8]
	 */
	public int[] swapEnds(int[] nums) {

		int first = 0;
		int last = nums.length-1;
		int tmp = 0;

		tmp = nums[first];
		nums[first] = nums[last];
		nums[last] = tmp;



		return nums;
	}

	/*
	 16. Given an array of ints, return true if the number of 1's is greater than the number of 4's
	 more14([1, 4, 1]) → true
	 more14([1, 4, 1, 4]) → false
	 more14([1, 1]) → true
	 */
	public boolean more14(int[] nums) {
		int count1 = 0;
		int count4 = 0;

		for (int i = 0; i < nums.length; i++) {
			if (nums[i] == 1) count1++;
			if (nums[i] == 4) count4++;

		}
		return (count1 > count4);
	}

	/*
	 17. Given a number n, create and return a new int array of length n, containing the numbers 0, 1, 2, ... n-1. 
	 The given n may be 0, in which case just return a length 0 array. You do not need a separate if-statement for 
	 the length-0 case; the for-loop should naturally execute 0 times in that case, so it just works. The syntax to 
	 make a new int array is: new int[desired_length]
	 fizzArray(4) → [0, 1, 2, 3]
	 fizzArray(1) → [0]
	 fizzArray(10) → [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
	 */
	public int[] fizzArray(int n) {
		int[] output = new int[n];

		for (int i = 0; i < output.length; i++) {
			output[i] = i;
		}

		return output;
	}

	/*
	 18. Given an array of ints, return true if every element is a 1 or a 4.
	 only14([1, 4, 1, 4]) → true
	 only14([1, 4, 2, 4]) → false
	 only14([1, 1]) → true
	 */
	public boolean only14(int[] nums) {

		for (int i = 0; i < nums.length; i++) {
			if (nums[i] != 1 && nums[i] !=4) return false;

		}
		return true;
	}
	
	/*
	 19. Given an array of ints, return true if it contains no 1's or it contains no 4's.
	 no14([1, 2, 3]) → true
	 no14([1, 2, 3, 4]) → false
	 no14([2, 3, 4]) → true
	 */
	public boolean no14(int[] nums) {

		for (int i = 0; i < nums.length; i++) {
			if (nums[i] == 1){
				for (int j = i + 1; j < nums.length; j++) {
					if (nums[j] == 4) return false;
				}
			}
		}
		return true;
	}

	/*
	 20. Given an array of ints, return true if there is a 1 in the array with a 2 somewhere later in the array.
	 has12([1, 3, 2]) → true
	 has12([3, 1, 2]) → true
	 has12([3, 1, 4, 5, 2]) → true
	 */
	public boolean has12(int[] nums) {

		for (int i = 0; i < nums.length; i++) {
			if (nums[i] == 1) {
				for (int j = i + 1; j < nums.length; j++) {
					if (nums[j] == 2) return true;
				}
			}
		}
		return false;
	}

	/*
	 21. Given start and end numbers, return a new array containing the sequence of integers from start up to but 
	 not including end, so start=5 and end=10 yields {5, 6, 7, 8, 9}. The end number will be greater or equal 
	 to the start number. Note that a length-0 array is valid. (See also: FizzBuzz Code)
	 fizzArray3(5, 10) → [5, 6, 7, 8, 9]
	 fizzArray3(11, 18) → [11, 12, 13, 14, 15, 16, 17]
	 fizzArray3(1, 3) → [1, 2]
	 */
	public int[] fizzArray3(int start, int end) {

		int[] output = new int[(end-start)];

		for (int i = 0; i < output.length; i++) {
			output[i] = start++;
		}

		return output;
	}

	/*
	 22. Given three ints, a b c, return true if one of them is 10 or more less than one of the others.
	 lessBy10(1, 7, 11) → true
	 lessBy10(1, 7, 10) → false
	 lessBy10(11, 1, 7) → true
	 */
	public boolean lessBy10(int a, int b, int c) {

		if(a+9 < b || a+9 < c) return true;
		if (b+9 < c || b+9 < a) return true;
		if (c+ 9 < a || c+9 < b) return true;
		return false;
	}

	/*
	 23. You have a green lottery ticket, with ints a, b, and c on it. If the numbers are all different from each other, 
	 the result is 0. If all of the numbers are the same, the result is 20. If two of the numbers are the same, the 
	 result is 10.
	 greenTicket(1, 2, 3) → 0
	 greenTicket(2, 2, 2) → 20
	 greenTicket(1, 1, 2) → 10
	 */
	public int greenTicket(int a, int b, int c) {

		if (a == b && b == c) return 20;
		if (a == b || b == c || a == c) return 10;
		return 0;
	}
	
	/*
	 24. Given 2 int values greater than 0, return whichever value is nearest to 21 without going over. Return 0 if they both 
	 go over.
	 blackjack(19, 21) → 21
	 blackjack(21, 19) → 21
	 blackjack(19, 22) → 19
	 */
	public int blackjack(int a, int b) {

		a = (a <= 21) ? a : 0;
		b = (b <= 21) ? b : 0;

		return Math.max(a, b);
	}

	/*
	 25. Given three ints, a b c, one of them is small, one is medium and one is large. Return true if the three values are evenly 
	 spaced, so the difference between small and medium is the same as the difference between medium and large.
	 evenlySpaced(2, 4, 6) → true
	 evenlySpaced(4, 6, 2) → true
	 evenlySpaced(4, 6, 3) → false
	 */
	public boolean evenlySpaced(int a, int b, int c) {

		List<Integer> findMiddle = new ArrayList<>();

		findMiddle.add(a);
		findMiddle.add(b);
		findMiddle.add(c);

		Collections.sort(findMiddle);


		return ( (findMiddle.get(2) - findMiddle.get(1)) == (findMiddle.get(1)) - findMiddle.get(0));
	}

}

File: m4-w10d1-js-intro-to-js-exercises/exercises.js

// 1. **SumDouble** Given two int values, return their sum. Unless the two values are the same, then return double their sum.

// sumDouble(1, 2) → 3
// sumDouble(3, 2) → 5
// sumDouble(2, 2) → 8

// function sumDouble(x, y) {
//     // do logic here
//     // return result;
//     return x + y;
// }		

function sumDouble(x, y) {
    sum = x + y;
    if (x === y) {
        return sum *= 2;
    }
    return sum;
}


// 2. **HasTeen** We'll say that a number is "teen" if it is in the range 13..19 inclusive. Given 3 int values, return true if 1 or more of them are teen.

// 		hasTeen(13, 20, 10) → true
// 		hasTeen(20, 19, 10) → true
// 		hasTeen(20, 10, 13) → true

function hasTeen(x, y, z) {
    if (x >= 13 && x <= 19) return true;
    if (y >= 13 && y <= 19) return true;
    if (z >= 13 && z <= 19) return true;
    return false;
}


// 3. **LastDigit** Given two non-negative int values, return true if they have the same last digit, such as with 27 and 57.

// 		lastDigit(7, 17) → true
// 		lastDigit(6, 17) → false
// 		lastDigit(3, 113) → true

function lastDigit(x, y) {
    var first = x.toString().split('').pop();
    var second = y.toString().split('').pop();

    if (first == second) { //loose equality
        return true;
    }
    return false;
}

// 4. **SeeColor** Given a string, if the string begins with "red" or "blue" return that color string, otherwise return the empty string.

// 		seeColor("redxx") → "red"
// 		seeColor("xxred") → ""
// 		seeColor("blueTimes") → "blue"

function seeColor(str) {
    var output = "";
    if (str.startsWith("red")) output = "red";
    if (str.startsWith("blue")) output = "blue";
    return output;
}


// 5. **MiddleThree** Given a string of odd length, return the string length 3 from its middle, so "Candy" yields "and". The string length will be at least 3.

// 		middleThree("Candy") → "and"
// 		middleThree("and") → "and"
// 		middleThree("solving") → "lvi"

function middleThree(str) {
    var output = "";

    output += str.charAt(str.length / 2 - 1);
    output += str.charAt(str.length / 2);
    output += str.charAt(str.length / 2 + 1);

    return output;


}

// 6. **FrontAgain** Given a string, return true if the first 2 chars in the string also appear at the end of the string, such as with "edited".

// 		frontAgain("edited") → true
// 		frontAgain("edit") → false
// 		frontAgain("ed") → true

function frontAgain(str) {
    var front = str.substring(0, 2);
    var back = str.charAt(str.length - 2) + str.charAt(str.length - 1);
    if (front == back) return true;
    return false;
}

// 7. **AlarmClock** Write a function that given a day of the week encoded as 0=Sun, 1=Mon, 2=Tue, ...6=Sat, and a boolean indicating if we are on vacation, displays the weeday name, and the time in the form of "7:00" indicating when the alarm clock should ring. Weekdays, the alarm should be "7:00" and on the weekend it should be "10:00". Unless we are on vacation -- then on weekdays it should be "10:00" and weekends it should be "off".

// 		alarmClock(1, false) → "Monday 7:00"
// 		alarmClock(5, true) → "Frday 10:00"
// 		alarmClock(0, false) → "Sunday 10:00"

function alarmClock(dayOfWeek, isVacation) {
    var alarmTime = "7:00";
    var dayOfWeekString = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"];
    var isWeekend;

    if (dayOfWeek == 0 || dayOfWeek == 6) isWeekend = true;
    if (isWeekend && isVacation) alarmTime = "off";
    else if (isWeekend) alarmTime = "10:00";
    else if (!isWeekend && isVacation) alarmTime = "10:00";

    return dayOfWeekString[dayOfWeek] + " " + alarmTime;

}

// 8. **MakeMiddle** Write a function that given an array of ints of even length, returns a new array length 2 containing the middle two elements from the original array. If the original array length is not even, or at least 2 elements in length, return an empty array.

// 		makeMiddle([1, 2, 3, 4]) → [2, 3]
// 		makeMiddle([7, 1, 2, 3, 4, 9]) → [2, 3]
// 		makeMiddle([1, 2]) → [1, 2]

function makeMiddle(arr) {
    var outputArray = [];
    if (arr.length % 2 == 1 || arr.length < 2) return outputArray;
    else {
        outputArray.push(arr[arr.length / 2 - 1]);
        outputArray.push(arr[arr.length / 2]);
    }
    return outputArray;
}

// 9. **OddOnly** Write a function that given an array of integer of any length, filters out the even number, and returns a new array of just the the odd numbers.

// 		oddOnly([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]) → [1, 3, 5, 7, 9, 11];
// 		oddOnly([2, 4, 8, 32, 256]); → []

function oddOnly(arr) {
    var outputArray = [ ];
    for (var i = 0; i < arr.length; i++) {
        if (arr[i] % 2 == 1) {
            outputArray.push(arr[i]);
        }
    }
    return outputArray;
    

}

// 10. **Weave** Write a function that given two arrays, interleaves the two arrays one element from each array at a time to return a new array made up of the interwoven elements of the original two arrays.

// 		weave([1, 3, 5], [2, 4]); → [1, 2, 3, 4, 5]
// 		weave([1, 3, 5], [2, 4, 6, 8]); → [1, 2, 3, 4, 5, 6, 8]

function weave(a, b){
    var outputArray =[];

    var largerArr = Math.max(a.length, b.length);

    for (var i = 0; i < largerArr; i++){
        if (i < a.length) outputArray.push(a[i]);
        if (i < b.length) outputArray.push(b[i]);
    }
    return outputArray;
}

// 11. **CigarParty** When squirrels get together for a party, they like to have cigars. A squirrel party is successful when the number of cigars is between 40 and 60, inclusive. Unless it is the weekend, in which case there is no upper bound on the number of cigars. Write a squirrel party function that return true if the party with the given values is successful, or false otherwise.

// 		cigarParty(30, false) → false
// 		cigarParty(50, false) → true
// 		cigarParty(70, true) → true

function cigarParty(cigars, isWeekend){
    
    return (isWeekend) ? (cigars >= 40) : (cigars >= 40 && cigars <=60);

}
// 12. **StringSplosion** Write a function that given a non-empty string like "Code" returns a string like "CCoCodCode". 

// 		stringSplosion("Code") → "CCoCodCode"
// 		stringSplosion("supercalifragilisticexpialidocious") → "ssusupsupesupersupercsupercasupercalsupercalisupercalifsupercalifrsupercalifrasupercalifragsupercalifragisupercalifragilsupercalifragilisupercalifragilissupercalifragilistsupercalifragilistisupercalifragilisticsupercalifragilisticesupercalifragilisticexsupercalifragilisticexpsupercalifragilisticexpisupercalifragilisticexpiasupercalifragilisticexpialsupercalifragilisticexpialisupercalifragilisticexpialidsupercalifragilisticexpialidosupercalifragilisticexpialidocsupercalifragilisticexpialidocisupercalifragilisticexpialidociosupercalifragilisticexpialidociousupercalifragilisticexpialidocious"

function stringSplosion(str){
    var output = "";
    for (var i = 1; i <= str.length; i ++){
    output += str.substring(0, i);
    }
    return output;
}
// 13. **FizzBuzz** Because you know you can't live without it, FizzBuzz. 


// 		fizzBuzz(3) → "Fizz"
// 		fizzBuzz(1) → 1
// 		fizzBuzz(10) → "Buzz"
// 		fizzBuzz(15) → "FizzBuzz"
// 		fizzBuzz(8) → 8

function fizzBuzz(num){
    var output = "";

    if (num %3 == 0) output += "Fizz";
    if (num %5 == 0) output += "Buzz";
    if (num %3 != 0 && num%5 != 0) output += num;
    return output;
}


// 14. **CountValues** Write a function that accepts an array of integer values, count of the number of times each value is found in the array, 
//and then display the values and their count.

// 		 countValues([1, 99, 43, 2, 55, 78, 99, 2345, 438, 2, 99, 107])  → 	
// 			1 : 1
// 			99 : 3
// 			43 : 1
// 			2 : 2
// 			55 : 1
// 			78 : 1
// 			2345 : 1
// 			438 : 1
// 			107 : 1

function countValues(arr){
    var map = new Map(0, 0);

    for (var i = 0; i < arr.length; i++){
        var count = 0;
        for (var j = i; j < arr.length; j++){
            if (arr[i] == arr[j]) {
                count++;
                console.log("Variable: " + arr[i] + "Count: " + count);
            }
            map.set(arr[i], count);
        }
    
    }
    return map;

}

// 15. **ReverseArray** Write an function that accepts an array of integer values, and display them in the reverse order they appeared in the array. One obvious solution is to simply loop through the integer array in reserve order, but see if you can come up with an alternative.

// 		reverseArray([1, 99, 43, 2, 55, 78, 99, 2345, 438, 2, 99, 107]) → 
// 		1 99 43 2 55 78 2345 438 107

function reverseArray(arr){
    outputArray = [];
    if (arr.length == 0) return outputArray;
    for (var i = 0; i <= arr.length; i++){
        while (arr.length > 0)
        outputArray.push(arr.pop());
    }
    return outputArray;
}

// 16. **BlackJack** Write a function that given 2 int values greater than 0, returns whichever value is nearest to 21 without going over. Return 0 if they both go over. 

// 		blackjack(19, 21) → 21
// 		blackjack(21, 19) → 21
// 		blackjack(19, 22) → 19

function blackJack(a, b){
    var a = (a <= 21) ? a : 0;
    var b= (b <= 21) ? b : 0;

    return Math.max(a, b);
}

// ## Challenge Exercises

// 1. **IQTest** Bob is preparing to pass an IQ test. The most frequent task in this test is to find out which one of the given numbers differs from the others. Bob observed that one number usually differs from the others in evenness. Help Bob — to check his answers, he needs a program that among the given numbers finds one that is different in evenness, and return the position of this number. _Keep in mind that your task is to help Bob solve a real IQ test, which means indexes of the elements start from 1 (not 0)_


// 		iqTest("2 4 7 8 10") → 3 //third number is odd, while the rest are even
// 		iqTest("1 2 1 1") → 2 // second number is even, while the rest are odd
// 		iqTest("") → 0 // there are no numbers in the given set
// 		iqTest("2 2 4 6") → 0 // all numbers are even, therefore there is no position of an odd number

// 2. **TitleCase** Write a function that will convert a string into title case, given an optional list of exceptions (minor words). The list of minor words will be given as a string with each word separated by a space. Your function should ignore the case of the minor words string -- it should behave in the same way even if the case of the minor word string is changed.


// * First argument (required): the original string to be converted.
// * Second argument (optional): space-delimited list of minor words that must always be lowercase except for the first word in the string. The JavaScript tests will pass undefined when this argument is unused.


// 		titleCase('a clash of KINGS', 'a an the of') // should return: 'A Clash of Kings'
// 		titleCase('THE WIND IN THE WILLOWS', 'The In') // should return: 'The Wind in the Willows'
// 		titleCase('the quick brown fox') // should return: 'The Quick Brown Fox'

// 3. **PerfectSquare** Complete the findNextSquare method that finds the next integral perfect square after the one passed as a parameter. Recall that an integral perfect square is an integer n such that sqrt(n) is also an integer. If the parameter is itself not a perfect square, than -1 should be returned. You may assume the parameter is positive.


// 		findNextSquare(121) → 144
// 		findNextSquare(625) → 676
// 		findNextSquare(114) → -1 // 114 is not a perfect square


